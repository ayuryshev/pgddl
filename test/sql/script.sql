\pset null _null_
\pset format unaligned

SELECT pg_ddlx_script('test_class_v'::regclass);
