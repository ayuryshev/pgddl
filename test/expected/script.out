\pset null _null_
\pset format unaligned
SELECT pg_ddlx_script('test_class_v'::regclass);
pg_ddlx_script
BEGIN;
-- SECTION DROP DEPENDANTS
/*
DROP VIEW test_class_v_opt2;
DROP VIEW test_class_v_opt1;
*/

-- SECTION MAIN

-- DROP VIEW test_class_v;

-- Type: VIEW ; Name: test_class_v; Owner: root

CREATE OR REPLACE VIEW test_class_v AS
 SELECT test_class_r.a,
    test_class_r.b,
    test_class_r.c,
    test_class_r.d,
    test_class_r.e,
    test_class_r.f,
    test_class_r.g,
    test_class_r.h,
    test_class_r.v
   FROM test_class_r;

COMMENT ON VIEW test_class_v IS NULL;

ALTER VIEW test_class_v SET (check_option='cascaded');

ALTER VIEW test_class_v OWNER TO root;
GRANT SELECT ON test_class_v TO PUBLIC;

-- SECTION CREATE DEPENDANTS

-- Type: VIEW ; Name: test_class_v_opt1; Owner: root

CREATE OR REPLACE VIEW test_class_v_opt1 AS
 SELECT test_class_v.a,
    test_class_v.b,
    test_class_v.c,
    test_class_v.d,
    test_class_v.e,
    test_class_v.f,
    test_class_v.g,
    test_class_v.h,
    test_class_v.v
   FROM test_class_v
  ORDER BY test_class_v.a;

COMMENT ON VIEW test_class_v_opt1 IS NULL;

ALTER VIEW test_class_v_opt1 SET (security_barrier='true');

ALTER VIEW test_class_v_opt1 OWNER TO root;

-- Type: VIEW ; Name: test_class_v_opt2; Owner: root

CREATE OR REPLACE VIEW test_class_v_opt2 AS
 SELECT test_class_v.a,
    test_class_v.b,
    test_class_v.c,
    test_class_v.d,
    test_class_v.e,
    test_class_v.f,
    test_class_v.g,
    test_class_v.h,
    test_class_v.v
   FROM test_class_v
  ORDER BY test_class_v.a;

COMMENT ON VIEW test_class_v_opt2 IS NULL;

ALTER VIEW test_class_v_opt2 SET (security_barrier='true');

ALTER VIEW test_class_v_opt2 OWNER TO root;
END;

(1 row)
